gateway_test.go

```diff
@@ -307,14 +307,14 @@ func testVirtualServerLookup(keys []string) (results []netip.Addr) {
 	return results
 }
 
-var testHTTPRouteIndexes = map[string][]netip.Addr{
+var testRouteIndexes = map[string][]netip.Addr{
 	"domain.gw.example.com": {netip.MustParseAddr("192.0.2.1")},
 	"shadow.example.com":    {netip.MustParseAddr("192.0.2.4")},
 }
 
-func testHTTPRouteLookup(keys []string) (results []netip.Addr) {
+func testRouteLookup(keys []string) (results []netip.Addr) {
 	for _, key := range keys {
-		results = append(results, testHTTPRouteIndexes[strings.ToLower(key)]...)
+		results = append(results, testRouteIndexes[strings.ToLower(key)]...)
 	}
 	return results
 }
```

Comment:
why renaming this? is this going to be generic for all routes?

Reply:
i re-use the same function for all GRPC, TLS and HTTP routes
i could separate them if you'd like but fundamentally, they only need an IP Address.

Reply:
sounds reasonable. thanks